### PR TITLE
[AMDGPU][test] Fix the wrong triples in lower-work-group-id-intrinsics-{hsa,pal}.ll. NFC

### PR DESCRIPTION
**Pull request by SixWeining**
- hsa -> amdhsa
- Use amdgcn-amd-amd{hsa,pal} for lower-work-group-id-intrinsics-{hsa,pal}.ll respectively

### CODE REVIEW

**Comment by SixWeining** [comment]
Seem the tests did not pass... :(

**Review by arsenm — COMMENTED** [review]
LGTM but the test will need update, this would have defaulted to unknown

**Comment by SixWeining** [comment]
> LGTM but the test will need update, this would have defaulted to unknown

lower-work-group-id-intrinsics-hsa.ll is the case but lower-work-group-id-intrinsics-pal.ll is not.

If use `-mtriple=amdgcn-amd-amdhsa`, we got error:
```
error: <unknown>:0:0: in function _amdgpu_cs_main void (): unsupported non-compute shaders with HSA
                                               
error: <unknown>:0:0: in function caller void (): unsupported non-compute shaders with HSA

llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h:910: void llvm::SIMachineFunctionInfo::setScratchRSrcReg(llvm::Register): Assertion `Reg != 0 && "S
hould never be unset"' failed.
```

I suppose  that lower-work-group-id-intrinsics-pal.ll only needs RUN lines with `-mtriple=amdgcn-amd-amdpal` while lower-work-group-id-intrinsics-hsa.ll only needs RUN lines with `-mtriple=amdgcn-amd-amdhsa`.

File: llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-hsa.ll [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=-architected-sgprs -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9,GFX9-SDAG %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=-architected-sgprs -global-isel=1 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9,GFX9-GISEL %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=+architected-sgprs -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9ARCH,GFX9ARCH-SDAG %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=+architected-sgprs -global-isel=1 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9ARCH,GFX9ARCH-GISEL %s
-; RUN: llc -mtriple=amdgcn-amd-amdpal -mcpu=gfx1200 -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX12,GFX12-SDAG %s
-; RUN: llc -mtriple=amdgcn-amd-amdpal -mcpu=gfx1200 -global-isel=1 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX12,GFX12-GISEL %s
+; RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx900 -mattr=-architected-sgprs -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9,GFX9-SDAG %s
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-hsa.ll:2]: The tests should not be deleted. If the error is emitted, the test file should be split up

Comment (SixWeining) [reply llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-hsa.ll:2]: Seems that had been split into llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-{hsa,pal}.ll.

**Review by arsenm — COMMENTED** [review]
Run lines probably still need to be updated, not removed. This is no longer stressing all the architected SGPR paths

File: llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-pal.ll [hunk 2]
```diff
@@ -1,38 +1,8 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=-architected-sgprs -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9 %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=-architected-sgprs -global-isel=1 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9 %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=+architected-sgprs -global-isel=0 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9ARCH-SDAG %s
-; RUN: llc -mtriple=amdgcn-amd-hsa -mcpu=gfx900 -mattr=+architected-sgprs -global-isel=1 -verify-machineinstrs < %s | FileCheck -check-prefixes=GFX9ARCH-GISEL %s
```
[the remark this answers is not in the document]
Comment (SixWeining) [reply llvm/test/CodeGen/AMDGPU/lower-work-group-id-intrinsics-pal.ll:5]: Yes, it should be. I have recovered them and changed to `amdgcn-amd-amdpal`. And same processing for `lower-work-group-id-intrinsics-hsa.ll`. Thanks.

**Review by arsenm — APPROVED** [review]

**Pull request merged**